### DISCUSSION TITLE
`OperationLike` trait

### DISCUSSION THREAD

**Discussion by raviqqe** [mlir-rs/melior#266] [Ideas]
- Currently, this applies to only modules.
- Do we really need this?

**Comment by Danacus** [comment]
Perhaps this would be useful if dialects weren't restricted to just constructors, but also provided accessors?

As an example, structs like `FuncOp` and `CallOp` could be created for the `func` dialects, and these could have accessors like `callee`. In this case, an `OperationLike` trait could be useful to convert from `Operation` to operation subtypes from different dialects. Maybe this is something to consider in the future?
